keps/sig-node/3759-kep-template/README.md

```diff
@@ -0,0 +1,1209 @@
+spec will be rejected by the control plane and all kubelets during
+unmarshalling. Workloads will have to be deleted and recreated with the old way
+of handling sidecars.
+
+The same problem will arise if the user wants to deploy such workloads on older
+clusters, or clusters without the feature flag enabled.
+
+
+### Version Skew Strategy
+
+Version skew is possible between the control plane and worker nodes as both
+should be aware of the new field used to flag sidecars inside initContainers.
+
+Therefore all cluster nodes, including master nodes, must be upgraded before the
+user can deploy sidecars using the new syntax.
+
+Also, since the feature flag applies to both kubelet and the control plane,
+similarly all cluster nodes need to have it enabled before deploying Pods with
+sidecars.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+Feature gate (also fill in values in kep.yaml)
+- Feature gate name: `SidecarContainers`
+- Components depending on the feature gate: kubelet, kube-apiserver
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [ ] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name:
+  - Components depending on the feature gate:
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
```

Comment:
should we put our feature name here?

Reply:
this item was referring to the fact that the config can be updated on the fly with DynamicKubeletConfig

Reply:
https://github.com/kubernetes/enhancements/pull/3764